cmd/jira/create.go

```diff
@@ -0,0 +1,100 @@
+package jira
+
+import (
+	"fmt"
+	"github.com/andygrunwald/go-jira"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"strings"
+)
+
+const (
+	TeamNameKey        = "jira_team"
+	TeamLabelKey       = "jira_team_label"
+	BoardIdLabel       = "jira_board_id"
+	DefaultDescription = ""
+	DefaultTicketType  = "Task"
+	DefaultProject     = "OSD"
+	SprintState        = "active"
+)
+
+var createCmd = &cobra.Command{
+	Use:   "create title",
+	Short: "creates a new ticket with the given name",
+	Long: `Creates a new ticket with the given name and a label specified by "jira_team_label" from the osdctl config.
+The ticket will be assigned to the caller and added to their team's current sprint as an OSD Task.
+A link to the created ticket will be printed to the console.`,
+	Example: `#Create a new Jira issue
+osdctl jira create "Update command to take new flag"`,
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		teamName := viper.GetString(TeamNameKey)
+		if teamName == "" {
+			return fmt.Errorf("%v value missing from config", TeamNameKey)
+		}
+
+		teamLabel := viper.GetString(TeamLabelKey)
+		if teamLabel == "" {
+			return fmt.Errorf("%v value missing from config", TeamLabelKey)
+		}
+
+		boardId := viper.GetInt(BoardIdLabel)
+		if boardId == 0 {
+			return fmt.Errorf("%v value missing from config", BoardIdLabel)
+		}
+
+		issue, err := CreateTicket(args[0], boardId, teamName, teamLabel)
+		if err != nil {
+			return fmt.Errorf("error creating ticket: %w", err)
+		}
+		fmt.Printf("Successfully created ticket:\n%v/browse/%v\n", utils.JiraBaseURL, issue.Key)
+		return nil
+	},
+}
+
+func CreateTicket(summary string, boardId int, teamName string, teamLabel string) (*jira.Issue, error) {
+	jiraClient, err := utils.GetJiraClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Jira client: %w", err)
+	}
+
+	user, _, err := jiraClient.User.GetSelf()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get jira user for self: %w", err)
+	}
+
+	issue, err := utils.CreateIssue(
+		jiraClient.Issue,
+		summary,
+		DefaultDescription,
+		DefaultTicketType,
+		DefaultProject,
+		user,
+		user,
+		[]string{teamLabel},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create issue: %w", err)
+	}
+
+	sprints, _, err := jiraClient.Board.GetAllSprintsWithOptions(boardId, &jira.GetAllSprintsOptions{State: SprintState})
+	if err != nil {
+		return nil, err
+	}
+
+	var activeSprint jira.Sprint
+	for _, sprint := range sprints.Values {
+		if strings.Contains(sprint.Name, teamName) {
+			activeSprint = sprint
+			break
+		}
+	}
+
+	_, err = jiraClient.Sprint.MoveIssuesToSprint(activeSprint.ID, []string{issue.ID})
```

Comment:
I imagine most tickets created w/ this command should be added to the backlog, no? Would it be worth adding a flag to toggle adding to current sprint vs backlogging?

Reply:
The use case we had during our sprint retro was that we had a lot of untracked work so we wanted to optimize to put this into the current sprint. I could get behind a `--skip-sprint` option that skips this step.

Reply:
I'm indifferent, however if we are to add a flag here I'd prefer the opposite of what's being suggested. That is - by default we don't add the ticket to the caller's current sprint, but we add a flag `--add-to-sprint` or something similar that would take this action.

Thoughts?

Reply:
+1 on default being no sprint.